client/client.go

```diff
@@ -179,18 +183,28 @@ func unmarshalRepoPackages(ctx context.Context, p, cacheDir string, cacheLife ti
 	if err != nil && !os.IsNotExist(err) {
 		return nil, err
 	}
-	logger.Infof("Fetching repo content for %s, cache either doesn't exist or is older than %v", p, cacheLife)
+	logger.Infof("Fetching repo content for %s, cache either doesn't exist or is older than %v", pName, cacheLife)
 
-	isGCSURL, bucket, object := goolib.SplitGCSUrl(p)
+	isGCSURL, bucket, object := goolib.SplitGCSUrl(pName)
 	if isGCSURL {
-		return unmarshalRepoPackagesGCS(ctx, bucket, object, p, cf, proxyServer)
+		return unmarshalRepoPackagesGCS(ctx, bucket, object, pName, cf, proxyServer)
 	}
-	return unmarshalRepoPackagesHTTP(p, cf, proxyServer)
+	return unmarshalRepoPackagesHTTP(ctx, p, cf, proxyServer)
 }
 
 // Get gets a url using an optional proxy server, retrying once on any error.
-func Get(path, proxyServer string) (*http.Response, error) {
-	httpClient := http.DefaultClient
+func Get(ctx context.Context, path, proxyServer string) (*http.Response, error) {
+	var httpClient *http.Client
```

Comment:
Prefer this to avoid the else:

httpClient := http.DefaultClient
if strings.HasPrefix(path, "oauth-") {
...
httpClient = oauth2.NewClient(ctx, creds.TokenSource)
}